web_widget_digital_signature/static/src/js/digital_sign.js

```diff
@@ -0,0 +1,131 @@
+                }
+            }
+            $(this.$el[0]).find('.clear_sign').click(function(){
+                self.$el.find('> img').remove();
+                images[self.name] = ""
+                $(self.$el[0]).find(".signature").show();
+                $(self.$el[0]).find(".signature").signature('clear');
+            });
+            $('.save_sign').click(function(){
+                var val
+                if($(self.$el[0]).find(".signature").hasClass( "kbw-signature" ) && ! $(self.$el[0]).find(".signature").signature('isEmpty')){
+                    $(self.$el[0]).find(".signature").hide();
+                    val = $(self.$el[0]).find(".signature > canvas")[0].toDataURL();
+                    images[self.name] = val.split(',')[1]
+                    var $img = $(QWeb.render("FieldBinaryImage-extend", { widget: self, url: val }));
+                    self.$el.find('> img').remove();
+                    self.$el.prepend($img);
+                    self.set('value',val.split(',')[1])
+                    var id = JSON.stringify(self.view.datarecord.id || null);
+                    var field = self.name;
+                    url = self.session.url('/web/binary/image', {
+                        model: self.view.dataset.model,
+                        id: id,
+                        field: field,
+                        t: (new Date().getTime()),
+                    });
+                }else{
+                        var id = JSON.stringify(self.view.datarecord.id || null);
+                        var field = self.name;
+                        if (self.options.preview_image)
+                            field = self.options.preview_image;
+                        url = self.session.url('/web/binary/image', {
+                                model: self.view.dataset.model,
+                                id: id,
+                                field: field,
+                                t: (new Date().getTime()),
+                        });
+                       var $img = $(QWeb.render("FieldBinaryImage-extend", { widget: self, url: url }));
+                       self.$el.find('> img').remove();
+                }
+            });
+            $img.load(function() {
+                if (! self.options.size)
+                    return;
+                $img.css("max-width", "" + self.options.size[0] + "px");
+                $img.css("max-height", "" + self.options.size[1] + "px");
+                $img.css("margin-left", "" + (self.options.size[0] - $img.width()) / 2 + "px");
+                $img.css("margin-top", "" + (self.options.size[1] - $img.height()) / 2 + "px");
+            });
+            $img.on('error', function() {
+                $img.attr('src', self.placeholder);
+                instance.webclient.notification.warn(_t("Image"), _t("Could not display the selected image."));
+            });
+        },
+    });
+    instance.web.FormView.include({
+        save: function(prepend_on_create) {
```

Comment:
this seems quite wrong to me. You break modularity here. Do whatever you need to do and call super() then.

But I don't believe you have to do anything here, you can also do that in Field.get_value()